sql/semisync_master.h

```diff
@@ -531,7 +563,7 @@ class Repl_semi_sync_master
    * Return:
    *  0: success;  non-zero: error
    */
-  int report_reply_binlog(uint32 server_id,
+  int report_reply_binlog(THD *replica_thd, uint32 server_id,
```

Comment:
A couple minor comments here:
  1) We don't need the whole `THD`; but we can pass the `thd->slave_info` because that is all that is needed.
  2) We can remove the `server_id` parameter; because it can be taken from `thd->slave_info.server_id`.